lib/src/main/java/io/ably/lib/http/Http.java

```diff
@@ -324,8 +325,10 @@ public void finalize() {
 	 * @return
 	 * @throws AblyException
 	 */
-	<T> T ablyHttpExecute(String path, String method, Param[] headers, Param[] params, RequestBody requestBody, ResponseHandler<T> responseHandler) throws AblyException {
+	public <T> T ablyHttpExecute(String path, String method, Param[] headers, Param[] params, RequestBody requestBody, ResponseHandler<T> responseHandler) throws AblyException {
 		int retryCountRemaining = Hosts.isRestFallbackSupported(this.host) ? options.httpMaxRetryCount : 0;
+		if(options.fallbackHosts != null)
+			Collections.shuffle(options.fallbackHosts);
```

Comment:
As above, we should really only need to shuffle this once and use that ordering for the rest of the lifetime of the instanced library